src/main/java/dinkplugin/notifiers/TradeNotifier.java

```diff
@@ -0,0 +1,160 @@
+package dinkplugin.notifiers;
+
+import dinkplugin.message.NotificationBody;
+import dinkplugin.message.NotificationType;
+import dinkplugin.message.templating.Replacements;
+import dinkplugin.message.templating.Template;
+import dinkplugin.notifiers.data.SerializedItemStack;
+import dinkplugin.notifiers.data.TradeNotificationData;
+import dinkplugin.util.ItemUtils;
+import dinkplugin.util.Utils;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.InventoryID;
+import net.runelite.api.Item;
+import net.runelite.api.ItemContainer;
+import net.runelite.api.annotations.Interface;
+import net.runelite.api.annotations.VarCStr;
+import net.runelite.api.events.WidgetClosed;
+import net.runelite.api.events.WidgetLoaded;
+import net.runelite.api.widgets.ComponentID;
+import net.runelite.client.callback.ClientThread;
+import net.runelite.client.game.ItemManager;
+import net.runelite.client.ui.DrawManager;
+import net.runelite.client.util.QuantityFormatter;
+import org.jetbrains.annotations.VisibleForTesting;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.awt.Image;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static dinkplugin.message.DiscordMessageHandler.PRIVATE_CHAT_WIDGET;
+
+@Slf4j
+@Singleton
+public class TradeNotifier extends BaseNotifier {
+    @VisibleForTesting
+    public static final String TRADE_ACCEPTED_MESSAGE = "Accepted trade.";
+    @VisibleForTesting
+    public static final @VarCStr int TRADE_COUNTERPARTY_VAR = 357;
+    private static final @Interface int TRADE_CONFIRMATION_GROUP = 334;
+
+    @Inject
+    private ClientThread clientThread;
+
+    @Inject
+    private DrawManager drawManager;
+
+    @Inject
+    private ItemManager itemManager;
+
+    private final AtomicReference<Image> image = new AtomicReference<>();
+
+    @Override
+    public boolean isEnabled() {
+        return config.notifyTrades() && super.isEnabled();
+    }
+
+    @Override
+    protected String getWebhookUrl() {
+        return config.tradeWebhook();
+    }
+
+    public void reset() {
+        image.lazySet(null);
+    }
+
+    public void onTradeMessage(String message) {
+        if (!TRADE_ACCEPTED_MESSAGE.equals(message) || !isEnabled()) {
+            this.reset();
+            return;
+        }
```

Comment:
Is there a risk of receiving an unrelated game message between taking the screenshot & the trade being accepted, causing the image to be reset?

Reply:
This shouldn't be an issue; we are only looking at `TRADE` message types so normal game messages won't matter

`TRADE` includes:
```
Player wishes to trade with you.
Sending trade offer...
Player is busy at the moment.
Other player declined trade.
Accepted trade.
```

(and when the trade interface is open, "wishes to trade with you" messages do not occur since the player is busy)